### PR TITLE
Fix #3507

### LINKED ISSUE #762: Expose floating-point BPM in MediaFile or normalize floats to ints in database
Running "beet write" causes updates of bpm values from integer to float (bpm: 113 -> 113.145) despite multiple runs of "beet write". "beet update -p" suggests same changes the other way around (bpm: 113.145 -> 113).

### LINKED ISSUE #3507: Integers are not normalized
### Problem

Beets currently allows the Model to set and the library to store any value for a field which is declared as `types.INTEGER` in the Item class. This behaviour however is not supported by MediaFile and for fields (such as bpm) which are [declared as integers](https://github.com/beetbox/mediafile/blob/master/mediafile.py#L1735)  these values are casted to integer before being written to the audio file tags. This results in the Model remaining always dirty and hence in the impossibility to have the library fully aligned with the media files.

This issue has been reported in #762 and discussed [here](https://discourse.beets.io/t/beet-write-bpm-quirk/738) on the forum. I am opening a new issue because even though this issue is most evident with the bpm field in reality it affects all fields declared `types.INTEGER` [in the Item class](https://github.com/beetbox/beets/blob/master/beets/library.py#L437).

In short, for any item in the library which has a bpm value in a non-integer format:
```text
     ...
     bpm: 119.434509277
     ...
```

Running this command:
```sh
$ beet write
```

will produce:

```
...
Madness - Full House: The Very Best of Madness - My Girl
  bpm: 119 -> 119.434509277
...
```

repeatedly. The audio file, if inspected through `beet info "My Girl"` will report:
```text
    ...
    bpm: 119
    ...
```
Therefore library and mediafile will remain forever out of sync.

### Setup

* OS: macOS 10.13.6
* Python version: 3.7.6
* beets version:  1.4.9 ([master](https://github.com/beetbox/beets/commit/8b8d223755662f499125b913ec9298faf3a2e3b2))
* Turning off plugins made problem go away (yes/no): no

My configuration (output of `beet config`) is:

```yaml
directory: /Volumes/Data/_TmpMusic_/
library: ~/.config/beets/test_library.db
asciify_paths: yes
id3v23: yes

ui:
    color: yes

plugins:
- the
- inline
- info
- fromfilename
- duplicates
- convert
- zero
- acousticbrainz

import:
    copy: no
    move: yes
    autotag: yes
    from_scratch: yes
    none_rec_action: asis
    duplicate_action: remove
    set_fields:
        albumstatus: new

paths:
    default: '%asciify{$initial_letter}/%the{%asciify{$albumartist}}/%asciify{$album}/$track. %asciify{$title}'
    singleton: V/Various Artists/%the{%asciify{$artist}}/$track. %asciify{$title}
    comp: V/Various Artists/%the{%asciify{$artist}}/$track. %asciify{$title}

musicbrainz:
    searchlimit: 3
item_fields:
    initial_letter: albumartist[0].upper() if albumartist else 'U'
zero:
    auto: yes
    fields:
    - month
    - day
    - encoder
    - lyrics
    - comments
    - artist_sort
    - original_month
    - original_day
    update_database: yes
    keep_fields: []
convert:
    command: ffmpeg -i $source -y -q:a 5 -f mp3 $dest
    extension: mp3
    max_bitrate: 182
    threads: 1
    auto: no
    dest: /Volumes/Data/_TmpMusic_/___converted___
    format: mp3
    formats:
        mp3: ffmpeg -i $source -y -q:a 5 -f mp3 $dest
        flac: ffmpeg -i $source -y -acodec flac -q:a 5 -f mp3 $dest
        aac:
            command: ffmpeg -i $source -y -vn -acodec aac -aq 1 $dest
            extension: m4a
        alac:
            command: ffmpeg -i $source -y -vn -acodec alac $dest
            extension: m4a
        opus: ffmpeg -i $source -y -vn -acodec libopus -ab 96k $dest
        ogg: ffmpeg -i $source -y -vn -acodec libvorbis -aq 3 $dest
        wma: ffmpeg -i $source -y -vn -acodec wmav2 -vn $dest
    pretend: no
    link: no
    hardlink: no
    id3v23: inherit
    tmpdir:
    quiet: no
    embed: yes

    paths: {}
    no_convert: ''
    never_convert_lossy_files: no
    copy_album_art: no
    album_art_maxwidth: 0
acousticbrainz:
    auto: no
    force: no
    tags: [bpm]
duplicates:
    album: no
    checksum: ''
    copy: ''
    count: no
    delete: no
    format: ''
    full: no
    keys: []
    merge: no
    move: ''
    path: no
    tiebreak: {}
    strict: no
    tag: ''
the:
    the: yes
    a: yes
    format: '{0}, {1}'
    strip: no
    patterns: []
pathfields: {}
album_fields: {}

```

### PR DESCRIPTION
**Pull request by adamjakab**
Adding normalize method to class `types.INTEGER` overriding the inherited method from the Type class that simply returned the provided value. Also provides Fix #762 and Fix #3507.

### CODE REVIEW

File: test/test_autotag.py [hunk 1]
```diff
@@ -91,7 +93,10 @@ def test_current_metadata_likelies(self):
                  for i in range(5)]
         likelies, _ = match.current_metadata(items)
         for f in fields:
-            self.assertEqual(likelies[f], '%s_1' % f)
+            if type(items[0]._fields[f]) in (Integer, PaddedInt):
```
Comment (sampsyo) [inline test/test_autotag.py]: Maybe a slightly easier way to do this would be `isinstance(likelies[f], int)`?

Comment (adamjakab) [reply test/test_autotag.py]: yes, it looks simpler.

**Comment by adamjakab** [comment]
> Nice work! This is really great!!

I am happy you think so. I was surprised to find an issue (#762) from 2014 which already brings this up. But TBH, I am being a bit selfish here in proposing this fix because I need this to work correctly for my new plugin. But hey! Isn't that always like that? ;) 

> I expected this to break more things, ...

Yes, me too ;) But there are not many tests targeting this. In fact one of the things I wanted to add here is some more tests - I'll look at the coverage and see if I can cover Integers.

> Would you mind adding a quick changelog entry?

Sure.

**Comment by adamjakab** [comment]
@sampsyo - this is ready to go!

**Comment by sampsyo** [comment]
Looks great! Thank you!!

**Pull request merged**